### PR TITLE
chore(docs): update docs and prompt to emphasize pushing of both yml files

### PR DESCRIPTION
**Pull request by huanjani**
Previously, only the `buildspec.yml` needed to be added to the source repo in order to build a pipeline. But now the buildspec checks the pipeline manifest to see which environment stages to deploy to (and only upgrades those envs), so both files need to be checked into the user's repo. This updates docs and CLI prompts to reflect that change. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: site/content/docs/concepts/pipelines.md [hunk 1]
```diff
@@ -93,27 +93,31 @@ stages:
...
+Typically, you'll update this file if you add new environments you want to deploy to, or want to track a different branch. The pipeline manifest is also where you may add a manual approval step before deployment or commands to run tests (see "Adding Tests," below) after deployment.
 
 ### Step 3: Updating the Buildspec (optional)
 
 Along with `pipeline.yml`, the `pipeline init` command also generated a `buildspec.yml` file in the `copilot/` directory. This contains the instructions for building and publishing your service. If you want to run any additional commands, besides `docker build`, such as unit tests or style checkers, feel free to add them to the buildspec's `build` phase.
 
 When this buildspec runs, it pulls down the version of Copilot which was used when you ran `pipeline init`, to ensure backwards compatibility.
 
-### Step 4: Creating your Pipeline
+### Step 4: Pushing New Files to your Repository
```
Comment (bvtujo) [inline site/content/docs/concepts/pipelines.md:104]: Can we reword this section to include the fact that we need *everything* in the copilot directory? Because we need service manifests, addons templates, pipeline and buildspec, and *especially* `.workspace`. 

edit: Efe got there first 🥇

File: internal/pkg/cli/pipeline_init.go [hunk 2]
```diff
@@ -205,13 +205,11 @@ func (o *initPipelineOpts) Execute() error {
...
-// RecommendedActions returns follow-up actions the user can take after successfully executing the command.
-func (o *initPipelineOpts) RecommendedActions() []string {
+// RequiredActions returns follow-up actions the user must take after successfully executing the command.
+func (o *initPipelineOpts) RequiredActions() []string {
 	return []string{
-		"Commit and push the generated buildspec and manifest file.",
-		fmt.Sprintf("Update the %s phase of your buildspec to unit test your services before pushing the images.", color.HighlightResource("build")),
-		"Update your pipeline manifest to add additional stages.",
-		fmt.Sprintf("Run %s to deploy your pipeline for the repository.", color.HighlightCode("copilot pipeline update")),
+		"Commit and push the generated buildspec and pipeline manifest files to the repository.",
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_init.go]: Should we just say 
```suggestion
		fmt.Sprintf("Commit and push the %s directory to your remote.", color.HighlightCode("copilot")),
```

File: internal/pkg/cli/pipeline_init.go [hunk 3]
```diff
@@ -205,13 +205,11 @@ func (o *initPipelineOpts) Execute() error {
...
+		fmt.Sprintf("Run %s to deploy your pipeline for the specified repository and branch.", color.HighlightCode("copilot pipeline update")),
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_init.go]: Maybe to be more concise?
```suggestion
		fmt.Sprintf("Run %s to create your pipeline.", color.HighlightCode("copilot pipeline update")),
```

File: internal/pkg/cli/pipeline_init.go [hunk 4]
```diff
@@ -205,13 +205,11 @@ func (o *initPipelineOpts) Execute() error {
...
 }
 
-// RecommendedActions returns follow-up actions the user can take after successfully executing the command.
-func (o *initPipelineOpts) RecommendedActions() []string {
+// RequiredActions returns follow-up actions the user must take after successfully executing the command.
+func (o *initPipelineOpts) RequiredActions() []string {
 	return []string{
-		"Commit and push the generated buildspec and manifest file.",
-		fmt.Sprintf("Update the %s phase of your buildspec to unit test your services before pushing the images.", color.HighlightResource("build")),
-		"Update your pipeline manifest to add additional stages.",
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_init.go:213]: Do we have this info written anywhere? I feel like this is helpful for customers to know what these files are for

```
✔ `copilot/buildspec.yml` to build your container images and run unit tests
✔ `copilot/manifest.yml` to add additional pipeline stages
Recommended follow-up actions:
- Commit and push ...
- Run copilot pipeline update ...
```

File: site/content/docs/concepts/pipelines.md [hunk 5]
```diff
@@ -24,14 +24,14 @@ Want to learn more about CodePipeline? Check out their [getting started docs](ht
 Creating a Pipeline requires only three steps:
 
 1. Preparing the pipeline structure.
-2. Committing the generated `buildspec.yml`.
+2. Committing and pushing the generated `pipeline.yml` and `buildspec.yml`.
```
Comment (efekarakus) [inline site/content/docs/concepts/pipelines.md]: The `.workspace` file needs to also be there to pick up the app name. So anything under `copilot/`

File: site/content/docs/concepts/pipelines.md [hunk 6]
```diff
@@ -24,14 +24,14 @@ Want to learn more about CodePipeline? Check out their [getting started docs](ht
...
 3. Creating the actual CodePipeline.
 
 Follow the three steps below, from your workspace root:
 
 ```bash
 $ copilot pipeline init
-$ git add copilot/buildspec.yml && git commit -m "Adding Pipeline Buildspec" && git push
+$ git add copilot/pipeline.yml copilot/buildspec.yml && git commit -m "Adding Pipeline Buildspec" && git push
```
Comment (bvtujo) [inline site/content/docs/concepts/pipelines.md]:
```suggestion
$ git add copilot/pipeline.yml copilot/buildspec.yml copilot/.workspace && git commit -m "Adding Pipeline Buildspec" && git push
```

File: internal/pkg/cli/pipeline_init.go (hunk 4 above)
Comment (huanjani) [reply internal/pkg/cli/pipeline_init.go:213]: Yes, we `	log.Infoln("The buildspec contains the commands to build and push your container images to your ECR repositories.")` when the file is created/confirmed to exist; I will add the bit about adding unit tests.

File: site/content/docs/concepts/pipelines.md (hunk 6 above)
Comment (huanjani) [reply site/content/docs/concepts/pipelines.md]: Okay to change to just `git add copilot &&`...?

**Comment by huanjani** [comment]
```
% copilot pipeline init
1st stage: test
Which repository would you like to use for your pipeline? codecommit::us-west-2://aws-sample
✔ Pipeline manifest file for aws-sample already exists at copilot/pipeline.yml, skipping writing it.
The manifest contains configurations for your CodePipeline resources, such as your pipeline stages and build steps.
Update the file to add additional stages, change the branch to be tracked, or add test commands or manual approval actions.
✔ Buildspec file for pipeline already exists at copilot/buildspec.yml, skipping writing it.
The buildspec contains the commands to build and push your container images to your ECR repositories.
Update the build phase to unit test your services before pushing the images.

Required follow-up actions:
- Commit and push the `copilot` directory to your repository.
- Run `copilot pipeline update` to create your pipeline.
jwh@38f9d33fed75 aws-copilot-sample-service %

File: site/content/docs/concepts/pipelines.md (hunk 6 above)
Comment (bvtujo) [reply site/content/docs/concepts/pipelines.md]: Hidden files like .workspace don't get picked up automatically by `git add copilot/*` so it needs to be explicitly added, unfortunately. 

Could do `git add copilot/* copilot/.workspace && ...`

Comment (bvtujo) [reply site/content/docs/concepts/pipelines.md]: or `git add --all copilot` also works!

Comment (huanjani) [reply site/content/docs/concepts/pipelines.md]: Ah, thanks. I'll change to your original suggestion bc it's the most clear for the reader.

File: internal/pkg/cli/pipeline_init.go (hunk 4 above)
Comment (huanjani) [reply internal/pkg/cli/pipeline_init.go:213]: See output below. Too wordy?

**Comment by huanjani** [comment]
This is what the output of `pipeline init` looked like previously (our of order!):
```
% copilot pipeline init
1st stage: test
2nd stage: quiz
Which repository would you like to use for your pipeline? codecommit::us-west-2://aws-sample
✔ Wrote the pipeline manifest for aws-sample at 'copilot/pipeline.yml'
The manifest contains configurations for your CodePipeline resources, such as your pipeline stages and build steps.
✔ Buildspec file for pipeline already exists at copilot/buildspec.yml, skipping writing it.
The buildspec contains the commands to build and push your container images to your ECR repositories.

Recommended follow-up actions:
- Commit and push the generated buildspec and manifest file.
- Update the build phase of your buildspec to unit test your services before pushing the images.
- Update your pipeline manifest to add additional stages.
- Run `copilot pipeline update` to deploy your pipeline for the repository.
%

File: internal/pkg/cli/pipeline_init.go [hunk 7]
```diff
@@ -505,7 +503,7 @@ func (o *initPipelineOpts) createPipelineManifest() error {
 		manifestMsgFmt = "Pipeline manifest file for %s already exists at %s, skipping writing it.\n"
 	}
 	log.Successf(manifestMsgFmt, color.HighlightUserInput(o.repoName), color.HighlightResource(manifestPath))
-	log.Infoln("The manifest contains configurations for your CodePipeline resources, such as your pipeline stages and build steps.")
+	log.Infoln("The manifest contains configurations for your CodePipeline resources, such as your pipeline stages and build steps.\nUpdate the file to add additional stages, change the branch to be tracked, or add test commands or manual approval actions.")
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_init.go]:
```suggestion
	log.Infof(`The manifest contains configurations for your CodePipeline resources, such as your pipeline stages and build steps.
Update the file to add additional stages, change the branch to be tracked, or add test commands or manual approval actions.
`)
```

File: internal/pkg/cli/pipeline_init.go [hunk 8]
```diff
@@ -545,7 +543,7 @@ func (o *initPipelineOpts) createBuildspec() error {
 		return err
 	}
 	log.Successf(buildspecMsgFmt, color.HighlightResource(buildspecPath))
-	log.Infoln("The buildspec contains the commands to build and push your container images to your ECR repositories.")
+	log.Infof("The buildspec contains the commands to build and push your container images to your ECR repositories.\nUpdate the %s phase to unit test your services before pushing the images.\n", color.HighlightResource("build"))
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_init.go]:
```suggestion
	log.Infof(`The buildspec contains the commands to build and push your container images to your ECR repositories.
Update the %s phase to unit test your services before pushing the images.
`, color.HighlightResource("build"))
```

File: site/content/docs/concepts/pipelines.md [hunk 9]
```diff
@@ -24,14 +24,14 @@ Want to learn more about CodePipeline? Check out their [getting started docs](ht
...
-2. Committing the generated `buildspec.yml`.
+2. Committing and pushing the generated files in the `copilot/` directory.
 3. Creating the actual CodePipeline.
 
 Follow the three steps below, from your workspace root:
 
 ```bash
 $ copilot pipeline init
-$ git add copilot/buildspec.yml && git commit -m "Adding Pipeline Buildspec" && git push
+$ git add copilot/pipeline.yml copilot/buildspec.yml copilot/.workspace && git commit -m "Adding Pipeline Buildspec" && git push
```
Comment (efekarakus) [inline site/content/docs/concepts/pipelines.md]:
```suggestion
$ git add copilot/pipeline.yml copilot/buildspec.yml copilot/.workspace && git commit -m "Adding pipeline artifacts" && git push
```

**Review by efekarakus — COMMENTED** [review]
Looks good just some minor formatting changes

**Review by efekarakus — APPROVED** [review]

File: site/content/docs/concepts/pipelines.md [hunk 10]
```diff
@@ -93,27 +93,31 @@ stages:
...
 
 Along with `pipeline.yml`, the `pipeline init` command also generated a `buildspec.yml` file in the `copilot/` directory. This contains the instructions for building and publishing your service. If you want to run any additional commands, besides `docker build`, such as unit tests or style checkers, feel free to add them to the buildspec's `build` phase.
 
 When this buildspec runs, it pulls down the version of Copilot which was used when you ran `pipeline init`, to ensure backwards compatibility.
 
-### Step 4: Creating your Pipeline
+### Step 4: Pushing New Files to your Repository
 
-Now that your `pipeline.yml` and `buildspec.yml` are created, check them in and push them to your repository. The `buildspec.yml` is needed for your pipeline's `build` stage to run successfully. Once you've done that, to actually create your pipeline run:
+Now that your `pipeline.yml`, `buildspec.yml`, and `.workspace` files have been created, add them to your repository. The entire `copilot/` directory is needed for your pipeline's `build` stage to run successfully.
```
Comment (Lou1415926) [inline site/content/docs/concepts/pipelines.md:106]: I am wondering If we need everything under the `copilot/` directory, should we also modify the instructions in "Creating a Pipeline in 3 steps"? 

It seems a little unclear to me whether we just need `pipeline.yml`, `buildspec.yml` and `.workspace` (since in the 3 steps instructions it only included these files), or we need the entire directory (as mentioned here)

**Review by Lou1415926 — COMMENTED** [review]
LGTM!! Just a question about the instruction ⬇️

File: site/content/docs/concepts/pipelines.md (hunk 10 above)
Comment (huanjani) [reply site/content/docs/concepts/pipelines.md:106]: Ah, good question! Will fix to make more specific!

**Comment by huanjani** [comment]
Sorry. Made a big git mess. Closing in favor of a new PR #1981.
